FATHMM-MKL uses protein sequence homology identified with HMMER3 to predict the impact of a missense variant. It uses SVM classifier to give prediction in p-values in the range [0,1]. Values above 0.5 are predicted to be deleterious, while those below 0.5 are predicted to be neutral or benign. It uses distinct predictors for positions either in coding regions and non-coding regions. The columns gathered are "chromosome  location  reference allele  alternative allele  nonCodingScore  codingScore".  
REFERENCE:https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4426838/pdf/btv009.pdf 
